__tests__/src/core/game.spec.js

```
@@ -0,0 +1,190 @@
+            expect(gridCopied.length).toEqual(grid.length);
+            expect(gridCopied).toEqual(expectedGrid);
+        });
+    });
+
+    describe('areGridsEquals', () => {
+        test('should return true if two grids are deep equals', () => {
+            const grid = [[1, 2, 3], [4, 5, 6], [7, 8, 0]];
+            const expectedGrid = [[1, 2, 3], [4, 5, 6], [7, 8, 0]];
+
+            const areEquals = Game.areGridsEquals(grid, expectedGrid);
+            expect(areEquals).toBe(true);
+        });
+
+        test('should return false if two grids are not deep equals', () => {
+            const grid = [[1, 2, 3], [4, 5, 6], [7, 8, 0]];
+            const expectedGrid = [[3, 12, 1], [5, 3, 6], [7, 8, 36]];
+
+            const areEquals = Game.areGridsEquals(grid, expectedGrid);
+            expect(areEquals).toBe(false);
+        });
+
+        test('should return false if the two grids do not have the same length', () => {
+            const grid = [[1, 2, 3], [4, 5, 6], [7, 8, 0]];
+            const expectedGrid = [[3, 12], [5, 3], [7, 8]];
+
+            const areEquals = Game.areGridsEquals(grid, expectedGrid);
+            expect(areEquals).toBe(false);
+        });
+    });
+
+    describe('findTileByValue', () => {
+        test('should find a tile that is in the grid', () => {
+            const grid = [[1, 2, 3], [4, 5, 6], [7, 8, 0]];
+            const value = 5;
+            const expectedCoords = new Coords(1, 1);
+
+            const coords = Game.findTileByValue(grid, value);
+
+            expect(coords.y).toEqual(expectedCoords.y);
+            expect(coords.x).toEqual(expectedCoords.x);
+            expect(grid[coords.y][coords.x]).toEqual(value);
+        });
+
+        test('should throw an error if the tile is not in the grid', () => {
+            const grid = [[1, 2, 3], [4, 5, 6], [7, 8, 0]];
+
+            expect(() => {
+                Game.findTileByValue(grid, 18);
+            }).toThrow();
```

This test can hide unexpected errors, it's a good practice to check for a particular type of error instead of any error.